torch/csrc/generic/Tensor.cpp

```diff
@@ -907,23 +905,66 @@ static THIndexTensor* THPTensor_(_calculateLinearIndices)(
   }
   THLongStorage_free(indexerSize);
 
+#ifdef THC_GENERIC_FILE
+  // Call GPU kernel for index calculation
+  THCudaLongTensor *cudaIndices =
+    THCudaLongTensor_newWithSize1d(LIBRARY_STATE indexingElements);
+  long baseOffset = THTensor_(storageOffset)(LIBRARY_STATE indexed);
+
+  // Need to pass broadcast Tensors to API, pass NULL ptr for all empty
+  // (i.e. not-advanced indexed) dims
+  std::vector<THCudaLongTensor *> indexers;
+  indexers.reserve(THTensor_(nDimension)(LIBRARY_STATE indexed.get()));
+
+  // Count the number of advanced indexers, and set the pointers to NULL for
+  // those that are not advanced indexing dims
+  unsigned int advancedIndexers = 0;
+  for (int i = 0; i < THTensor_(nDimension)(LIBRARY_STATE indexed.get()); ++i) {
+    if (flattenedBroadcasters.count(i) > 0) {
+      ++advancedIndexers;
+    } else {
+      indexers[i] = NULL;
```

Comment:
This is incorrect. You either need to `reserve` and then use `push/emplace` to fill the container, or pass the size to the constructor (which will fill the container with this many default values - NULL in this case) and later access the elements randomly.